agent/fishing_agent.py

```
@@ -492,19 +539,30 @@ def main_loop(self) -> bool:
+        # 且 PipelineTask.cpp 每次命中节点都会重置 error_handling,框架自带的
+        # "error handling loop detected" 保护因此不会触发),所以必须另设 wall-clock 上界。
+        # 注意:它只能在轮与轮之间生效,拦不住单次 run_task 内部的阻塞 —— 那要靠切断
+        # pipeline 自环,属另一处待定改动。
+        deadline = time.monotonic() + max_seconds
+
         try:
             while self.running and not self.context.tasker.stopping:
                 if max_count and self.fish_count >= max_count:
                     break
+                if time.monotonic() >= deadline:
+                    print(f"warn: ⏱️ 已达总时长上限 {max_seconds:.0f}s，结束钓鱼")
```

Ruff (0.16.1)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)